be/src/exec/hash_join_node.cpp

```diff
@@ -109,6 +109,9 @@ Status HashJoinNode::init(const TPlanNode& tnode, RuntimeState* state) {
             }
             DCHECK(match_exactly_once);
         }
+    } else {
```

Comment:
add some comment why colocate HJ cases needs this

Reply:
Resoloved.